src/extension/rn-extension.ts

```diff
@@ -5,64 +5,90 @@ import {FileSystem} from "../common/node/fileSystem";
-                        () => commandPaletteHandler.runIos()));
-                    context.subscriptions.push(vscode.commands.registerCommand("reactNative.startPackager",
-                        () => commandPaletteHandler.startPackager()));
-                    context.subscriptions.push(vscode.commands.registerCommand("reactNative.stopPackager",
-                        () => commandPaletteHandler.stopPackager()));
+const reactNativeProjectHelper = new ReactNativeProjectHelper(vscode.workspace.rootPath);
+const fsUtil = new FileSystem();
 
-                    const nodeDebugPath = vscode.extensions.getExtension("andreweinand.node-debug").extensionPath;
-                    const fsUtil = new FileSystem();
-                    fsUtil.writeFile(path.resolve(__dirname, "../", "debugger", "nodeDebugLocation.json"), JSON.stringify({ nodeDebugPath })).done();
-                });
-        }).done();
+export function activate(context: vscode.ExtensionContext): void {
+    entryPoint.runApp("react-native", () => <string>require("../../package.json").version, "Failed to activate the React Native Tools extension", () => {
+        return reactNativeProjectHelper.isReactNativeProject()
+            .then(isRNProject => {
+                if (isRNProject) {
+                    warnWhenReactNativeVersionIsNotSupported();
+                    entryPoint.runFunction("debugger.setupLauncherStub", "Failed to setup the stub launcher for the debugger", () =>
+                        setupReactNativeDebugger().then(() =>
+                            setupReactDir(context)), /*areErrorsFatal*/ false);
+                    entryPoint.runFunction("intelliSense.setup", "Failed to setup IntelliSense", () =>
+                        IntellisenseHelper.setupReactNativeIntellisense(), /*areErrorsFatal*/ false);
+                }
+                entryPoint.runFunction("debugger.setupNodeDebuggerLocation", "Failed to configure the node debugger location for the debugger", () =>
+                    configureNodeDebuggerLocation(), /*areErrorsFatal*/ false);
+                registerReactNativeCommands(context);
+            });
+    });
 }
 
 export function deactivate(): void {
     // Kill any packager processes that we spawned
-    commandPaletteHandler.stopPackager();
+    entryPoint.runFunction("extension.deactivate", "Failed to stop the packager while closing React Native Tools",
+        () => commandPaletteHandler.stopPackager(), /*areErrorsFatal*/ true);
+}
+
+function configureNodeDebuggerLocation(): Q.Promise<void> {
+    const nodeDebugPath = vscode.extensions.getExtension("andreweinand.node-debug").extensionPath;
+    return fsUtil.writeFile(path.resolve(__dirname, "../", "debugger", "nodeDebugLocation.json"), JSON.stringify({ nodeDebugPath }));
+}
+
+function setupReactDir(context: vscode.ExtensionContext): Q.Promise<void> {
+    const reactDirManager = new ReactDirManager();
+    let createdDir = reactDirManager.create();
```

Comment:
What are you trying to achieve with the `createdDir`? I don't see you using the return value anywhere.